maintenance/wikia/discussion-alerts.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * Alerts for discussion content.
+ *
+ * @author tor
+ * @see SUS-2420
+ *
+ * This script should be run for wiki_id 177 where $wgDiscussionAlertsQueries WikiFactory variable is defined
+ */
+
+require_once __DIR__ . '/../commandLine.inc';
+
+echo "Init...\n";
+
+$db = 'specials';
+$perpage = 1000;
+$date = gmDate( "Y-m-d\TH:i:s\Z", strtotime( '-30 days' ) );
+
+
+$actualQuery = '(title_en:"PATTERN" OR nolang_en:"PATTERN") AND touched:[' . $date . ' TO *]';
+
+$searchUrl = "http://prod.search-master.service.sjc.consul:8983/solr/main/select/?wt=php&fl=ns,wid,pageid,url,host,titleStrict&q=" . urlencode( $actualQuery ) . "&rows=$perpage&start=";
+
+$queries = $wgDiscussionAlertsQueries;
+$found = array();
+$body = 'This message contains pages containing phrases which are marked as to be tracked (in $wgDiscussionAlertsQueries).' . "\n\nThis listing contains pages edited no earlier than $date (30 days ago) and not previously reported.\n\nPlease ensure the end line is visible before closing this ticket.  If not, request a full copy from Community Engineering.";
+
+echo "Connecting to DB...\n";
+
+$dbr = wfGetDB( DB_SLAVE, array(), $db );
+$dbw = wfGetDB( DB_MASTER, array(), $db );
+
+$threadNamespaces = [ NS_USER_WALL_MESSAGE, NS_WIKIA_FORUM_BOARD_THREAD ];
+
+// Step 1: process queries and fetch data from Solr
+foreach ( $queries as $query ) {
+
+	echo "Processing query: $query\n";
+
+	// initializing variables
+	$i = 1;
+	$found[$query] = array();
+
+	// prepare solr URL
+	$finalUrl = str_replace( 'PATTERN', urlencode( $query ), $searchUrl );
+
+	$body .= "\n\n== $query ==\n\n";
+
+	$threadReported = [];
+
+	while ( $i + $perpage < $numFound || $i == 1 ) {
+		echo "Fetching result...\n";
+		$result_raw = file_get_contents( $finalUrl . $i );
+
+		if ( empty( $result_raw ) ) {
+			die( "Something went horribly wrong. No response from Solr.\n" );
+		} else {
+			echo "Processing result...\n";
+			eval("\$result = " . $result_raw . ";" );
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove this use of the "eval" function. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1523)